PHOENIX-6125 : Disable region split for SYSTEM.TASK

virajjasani opened the pull request.

stoty commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  37m 31s |  master passed  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 21s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 47s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 58s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m 39s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 18s |  phoenix-core: The patch generated 11 new + 2241 unchanged - 1 fixed = 2252 total (was 2242)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  7s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  96m 19s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   | 183m 28s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.SystemTablesCreationOnConnectionIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/923 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 21d3231bbcc3 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / fe7c46c |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/1/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/1/testReport/ |
| Max. process+thread count | 6981 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

stoty commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| -1 :x: |  mvninstall  |  37m 31s |  root in master failed.  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 46s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 59s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |  32m 50s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 23s |  phoenix-core: The patch generated 1 new + 2241 unchanged - 1 fixed = 2242 total (was 2242)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 12s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  96m 33s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 28s |  The patch does not generate ASF License warnings.  |
|  |   | 182m 43s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/923 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 9edc0122b54a 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 628fa0d |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/testReport/ |
| Max. process+thread count | 6943 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

ChinmaySKulkarni commented on phoenix-core/src/it/java/org/apache/phoenix/end2end/DisableSplitForTaskTableIT.java:
```diff
@@ -0,0 +1,134 @@
+                .getRegionSplitPolicyClassName();
+            assertEquals(DisabledRegionSplitPolicy.class.getName(),
+                taskSplitPolicy);
+        }
```
Do we really need a separate IT for this? There is an existing IT that uses the upgrade counting driver and services. Let's reuse that instead

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -1336,6 +1337,8 @@ private TableDescriptor ensureTableCreated(byte[] physicalTableName, PTableType
                     newDesc.setRegionSplitPolicyClassName(IndexRegionSplitPolicy.class.getName());
                 }
+                // disable split policy for SYSTEM.TASK
+                isSplitPolicyUpdatedForTaskTable(newDesc);
```
Since we have the namespace mapping enabled property available at this point, can't we get the exact expected name i.e. SYSTEM.TASK vs SYSTEM:TASK instead of trying both variations inside this method?

ChinmaySKulkarni replied:
All table creations go through this method so instead of polluting this with SYSTEM.TASK specific stuff, I think we should divide the problem into 2 parts:

1. Fresh clusters -> Change the DDL for SYSTEM.TASK (CQSI.CREATE_TASK_METADATA) itself so this would work directly i.e. add
HTableDescriptor.SPLIT_POLICY + "='" + SystemTaskSplitPolicy.class.getName() + "',\n"
and introduce a new class `SystemTaskSplitPolicy` which is equivalent to `DisabledRegionSplitPolicy` for now, but can be easily changed in the future if required without unloading and reloading the coproc.

2. Upgrade path -> <=4.15 clusters would go through upgradeSystemTask() and here we could load the correct split policy. That way, we don't need the change inside ensureTableCreated() and it is much cleaner. One corner case here (highly unlikely, but still possible) is if a cluster already has SYSTEM.TASK split. In that case, we might want to throw an exception or maybe merge regions of SYSTEM.TASK

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -1451,7 +1454,35 @@ private TableDescriptor ensureTableCreated(byte[] physicalTableName, PTableType
     }
-    
+
+    private boolean isSplitPolicyUpdatedForTaskTable(
```
Since this method actually also replaces the split policy, we should rename this method to indicate that.

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -1451,7 +1454,35 @@ private TableDescriptor ensureTableCreated(byte[] physicalTableName, PTableType
-    
+
+    private boolean isSplitPolicyUpdatedForTaskTable(
+            final TableDescriptorBuilder tdBuilder) {
```
Can we add a new unit test for this method?

virajjasani replied:
Done

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -4006,6 +4037,33 @@ private PhoenixConnection upgradeSystemTask(PhoenixConnection metaConnection)
+                try {
+                    tableName = TableName.valueOf(tableNameStr);
+                    td = admin.getDescriptor(tableName);
+                } catch (org.apache.hadoop.hbase.TableNotFoundException tnfe) {
```
Same question here, can we not use namespace mapping property to ensure we are looking for the correct name '.' vs ':'?

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -4006,6 +4037,33 @@ private PhoenixConnection upgradeSystemTask(PhoenixConnection metaConnection)
+                        tableDescriptorBuilder.build(), tableName.getName());
+                }
+            } catch (InterruptedException | TimeoutException ite) {
+                throw new SQLException(PhoenixDatabaseMetaData.SYSTEM_TASK_NAME
```
Do we have any specific variant of SQLException to handle any exceptions thrown during the upgrade process?

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -4006,6 +4037,33 @@ private PhoenixConnection upgradeSystemTask(PhoenixConnection metaConnection)
             }
+            // If SYSTEM.TASK does not have disabled regions split policy,
+            // set it up here while upgrading it
+            try (Admin admin = metaConnection.getQueryServices().getAdmin()) {
```
Also, it would be better to change the DDL statement (CQSI.CREATE_TASK_METADATA) to reflect the new SplitPolicy:
HTableDescriptor.SPLIT_POLICY + "='" + SystemTaskSplitPolicy.class.getName() + "',\n"
and introduce a new class `SystemTaskSplitPolicy` which is equivalent to `DisabledRegionSplitPolicy` for now, but can be easily changed in the future if required without unloading and reloading the coproc.

ChinmaySKulkarni replied:
So new clusters would always come up with the correct split policy for SYSTEM.TASK and <=4.15 clusters would go through this path (upgradeSystemTask()) and loads the correct split policy. That way, we don't need the change inside ensureTableCreated() and it is much cleaner.

One corner case (highly unlikely, but still possible) is if a cluster already has SYSTEM.TASK split. In that case, we might want to throw an exception or maybe merge regions of SYSTEM.TASK

ChinmaySKulkarni reviewed (changes requested):
Added some comments. Also, the test failure seems related since it is complaining about TableNotFoundException for SYSTEM.TASK

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -1451,7 +1454,35 @@ private TableDescriptor ensureTableCreated(byte[] physicalTableName, PTableType
+        if (tdBuilder.build().getTableName().getNameAsString()
+                .equals(taskTable)) {
+            isTaskTable = true;
+        } else {
```
Can't we use the actual namespace mapping property to get the exact expected name i.e. SYSTEM.TASK vs SYSTEM:TASK instead of trying both variations?

ChinmaySKulkarni commented on phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java:
```diff
@@ -4006,6 +4037,33 @@ private PhoenixConnection upgradeSystemTask(PhoenixConnection metaConnection)
             }
+            // If SYSTEM.TASK does not have disabled regions split policy,
+            // set it up here while upgrading it
+            try (Admin admin = metaConnection.getQueryServices().getAdmin()) {
```
Is it possible to add a test case inside BackwardsCompatibilityIT to make sure that the upgrade path changes work and SYSTEM.TASK comes up with the expected split policy?

virajjasani replied:
Added a test on `BackwardsCompatibilityIT`. Tested on 4.x with compatible client versions: 4.14.3 and 4.15.0.
For one, split policy was set by DDL statement at server and for another, upgrade path was followed to ensure split policy is updated.

virajjasani commented:
> Added some comments. Also, the test failure seems related since it is complaining about TableNotFoundException for SYSTEM.TASK

Thanks @ChinmaySKulkarni . Addressed concerns and updated PR accordingly. Test failure is also taken care of.

stoty commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| -1 :x: |  mvninstall  |  36m 22s |  root in master failed.  |
| +1 :green_heart: |  compile  |   0m 59s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 33s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 55s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |  32m 23s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 37s |  phoenix-core: The patch generated 34 new + 2543 unchanged - 33 fixed = 2577 total (was 2576)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 9 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 1 line(s) with tabs.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  3s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  94m 19s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   | 179m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/923 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 92687e421e34 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 628fa0d |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/whitespace-eol.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/whitespace-tabs.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/testReport/ |
| Max. process+thread count | 6844 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

stoty commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| -1 :x: |  mvninstall  |  36m  0s |  root in master failed.  |
| +1 :green_heart: |  compile  |   0m 59s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 33s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 49s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |  32m 35s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 34s |  phoenix-core: The patch generated 34 new + 2542 unchanged - 33 fixed = 2576 total (was 2575)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 9 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 1 line(s) with tabs.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  3s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  95m  6s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   | 179m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/923 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux fc540289043d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 628fa0d |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/whitespace-eol.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/whitespace-tabs.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/testReport/ |
| Max. process+thread count | 6699 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/6/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

stoty commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| -1 :x: |  mvninstall  |   3m 40s |  root in master failed.  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 44s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 51s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   1m  4s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 51s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 51s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 50s |  phoenix-core: The patch generated 41 new + 3120 unchanged - 35 fixed = 3161 total (was 3155)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 9 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 1 line(s) with tabs.  |
| +1 :green_heart: |  javadoc  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  2s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  96m 58s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   | 117m 58s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/923 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux bef3c8691c66 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / f0e5f05 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/branch-mvninstall-root.txt |
| mvninstall | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/patch-mvninstall-root.txt |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/whitespace-eol.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/whitespace-tabs.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/testReport/ |
| Max. process+thread count | 6605 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/7/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

virajjasani commented on phoenix-core/src/it/java/org/apache/phoenix/end2end/SystemTablesUpgradeIT.java:
```diff
@@ -25,61 +25,68 @@
             countUpgradeAttempts++;
         }
-        
+
```
These changes are specific only to whitespace issues that precommit build reported.

stoty commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  36m 16s |  master passed  |
| +1 :green_heart: |  compile  |   0m 55s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 49s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 54s |  phoenix-core in master has 973 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 51s |  phoenix-core: The patch generated 10 new + 3151 unchanged - 4 fixed = 3161 total (was 3155)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  2s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  95m 18s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   | 180m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/9/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/923 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 2fb337a3e03c 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 5b58d9b |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/9/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/9/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/9/testReport/ |
| Max. process+thread count | 6735 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-923/9/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

ChinmaySKulkarni reviewed (commented):
+1 thanks @virajjasani I will commit this to master and 4.x today

ChinmaySKulkarni commented:
@virajjasani the master patch doesn't apply cleanly to 4.x so please put up a patch/PR for 4.x, thanks

virajjasani commented:
Done, uploaded patch on Jira. Thanks for the review @ChinmaySKulkarni
